capstone/src/test/java/com/google/sps/servlets/CommentsServletTest.java

```diff
@@ -0,0 +1,152 @@
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.google.sps.Objects.Post;
+import com.google.sps.Objects.Comment;
+
+public class CommentsServletTest {
+
+  private static final String USER_EMAIL = "test@test.com";
+  private static final String USER_ID = "test";
+  private static final String AUTHOR_ID = "123123123";
+  private static final String POST_TEXT = "a great post";
+  private static final String CHALLENGE_NAME = "run 4 miles";
+  private static final String IMG = "";
+  private static final long TIMESTAMP = 123123123;
+  private static final long POST_ID = 1;
+  private static final String COMMENT_TEXT = "this is a comment";
+
+private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig()
+                  .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+              new LocalUserServiceTestConfig())
+          .setEnvEmail(USER_EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com")
+          .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", USER_ID)));
+
+  private final Post POST_1 = new Post(
+    POST_ID, /* postId */ 
+    AUTHOR_ID, /* authorId */ 
+    POST_TEXT, /* postText */
+    new ArrayList<Comment>(), /* comments */
+    CHALLENGE_NAME, /* challengeName */
+    TIMESTAMP, /* timestamp */
+    IMG, /* img */
+    new HashSet<String>() /* likes */);
+
+  private final Comment COMMENT_1 = new Comment (
+    System.currentTimeMillis(), /* timestamp */
+    COMMENT_TEXT, /* commentText */
+    USER_ID /* userId */
+  );
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DatastoreService datastore;
+  private CommentsServlet commentServlet;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+    commentServlet = new CommentsServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    responseWriter = null;
+    datastore = null;
+    commentServlet = null;
+  }
+
+  private void populateDatabase(DatastoreService datastore) {
+    // Add test data.
+    datastore.put(POST_1.createPostEntity());
+  }
+
+  @Test
+  public void doPost_addComment() throws Exception {
+    populateDatabase(datastore);
```

Comment:
You should call this once in the setUp instead of at the beginning of every test